cumulus/tasks/sf-sns-report/index.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const get = require('lodash.get');
+const { StepFunction } = require('@cumulus/ingest/aws');
+const { setGranuleStatus, sns } = require('@cumulus/common/aws');
+const errors = require('@cumulus/common/errors');
+const cumulusMessageAdapter = require('@cumulus/cumulus-message-adapter-js');
+
+/**
+ * Determines if there was a valid exception in the input message
+ *
+ * @param {Object} event - aws event object
+ * @returns {boolean} true if there was an exception, false otherwise
+ */
+function eventFailed(event) {
+  if (event.exception) {
+    if (typeof event.exception === 'object') {
+      // this is needed to avoid flagging cases like "exception: {}" or "exception: 'none'"
+      if (Object.keys(event.exception).length > 0) {
+        return true;
+      }
+    }
+  }
+  // Error and error keys are not part of the cumulus message
+  // and if they appear in the message something is seriously wrong
+  else if (event.Error || event.error) {
```

Comment:
Do we know an instance of this occurring? A better option might be to whitelist the top-level keys we expect from a cumulus message, otherwise this seems unnecessary

Reply:
I don't.  I don't know if removing this logic will break something.

Reply:
this was reported in an earlier bug report and this is the fix for that bug.

Reply:
ah ok - do you recall if that bug was before or after cumulus message adapter integration?

Reply:
@abarciauskas-bgse brings up a very good point.  When you're using the message adapter, the only keys that the simplified event are going to have are "config" and "input".  This is never going to trigger.